autoprotocol/protocol.py

```diff
@@ -4979,8 +4979,9 @@ def provision(self, resource_id, dests, volumes):
                     d.container):
                 self.instructions[-1].to.append(xfer)
             else:
-                return self._append_and_return(Provision(resource_id,
+                self._append_and_return(Provision(resource_id,
                                                          dest_group))
+        return self
```

Comment:
didn't spot this, but generally we return the generated Instruction here, and not the protocol

Reply:
Actually there is no need to return as the method invocation is on the same Protocol object. I have validated it and works fine.